testcases/cloud_user/elb/ssl_termination.py

```diff
@@ -125,30 +127,199 @@ def ssl_termination(self):
         frontend_cert_name = "elb-ssl-test-"+str(int(time.time()))
         self.tester.add_server_cert(cert_name=frontend_cert_name)
 
-        # remove listener on port 80 and create a new listener on HTTPS port 443
-        self.tester.remove_lb_listener(lb_name=self.load_balancer.name, port=self.load_balancer_port)
-        self.update_listener(lb_port=443,
-                             lb_protocol="HTTPS",
-                             instance_port=80,
-                             instance_protocol="HTTP",
-                             cert=frontend_cert_name)
+        # remove any existing listeners
+        self.remove_all_listeners(lb_name=self.load_balancer.name)
+
+        # get the arn of the certificate
+        frontend_cert_arn = self.tester.get_server_cert(frontend_cert_name).arn
+
+        self.tester.update_listener(lb=self.load_balancer,
+                                    lb_port=443,
+                                    lb_protocol="HTTPS",
+                                    instance_port=80,
+                                    instance_protocol="HTTP",
+                                    cert_arn=frontend_cert_arn)
 
         # perform https requests to LB
         self.tester.sleep(10)
         self.tester.generate_http_requests(url=lb_url, count=10)
 
-    def update_listener(self, lb_port, lb_protocol, instance_port, instance_protocol, cert):
-        self.debug("Configuring ELB listener")
+    def end_to_end_ssl_termination(self):
```

Comment:
@tbeckham First of all, great work! Not a big issue as long as it does the right thing, but I think you meant end_to_end_ssl_authentication instead of *_ssl_termination, yes?